### DISCUSSION TITLE
Does it support V21.0?

### DISCUSSION THREAD

**Discussion by wbefk** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#104] [Q&A]
Hi,
I am very new to the WhatsApp Business Cloud API and have struggled with facebook/meta documentation/user "experience" and with routing as some users before.
I created my sample app based on your sample `WhatsAppBusinessCloudAPI.Web` and managed to get it working (sending text messages, receiving messages thru webhook).
What does not work is the account metrics part. I found out that in `Analytics.cshtml` and `ConversationAnalytics.cshtml` the action `HomeController.QRCodeMessageList` is called while it should be `HomeController.Analytics` resp. `HomeController.ConversationAnalytics`. But then the app throws an error of 
> Unsupported get request. Please read the Graph API documentation at https://developers.facebook.com/docs/graph-api. 

I couldn't find such a documentation for v21.0. 
* Is v21.0 not supported?
* Also the `WhatsAppBusinessRequestEndpoint.BaseAddress` is ending with "v.19.0/". Is it possible to set the version as needed?
* The api documentation also mentions a video url of "graph-video.facebook.com". I can't find this url in the WhatsAppBusiness.CloudApi.

Btw. Thank you very much for your wonderful work on this API. It will help a lot.

**Comment by gabrieldwight** [comment]
Hi, i'm currently testing V21. 

I'm still figuring out how to add support for setting graph API versions.

Is the video URL returned after uploading a video to cloud API?

**Reply by wbefk** [reply]
facebook documentation seems to change faster than developers can read it ;), thus becoming inconsistent :(. "graph-video.facebook.com" is deprecated in favor of  "graph.facebook.com", see https://developers.facebook.com/docs/video-api/getting-started/.

If you receive a message via webhook, only media ids are returned. You then have to obtain a media url and then you can download the media. It would be nice to support this process in the Whatsapp-Business-Cloud_Api-Net.

**Reply by gabrieldwight** [reply]
Getting media url from media Id is already supported as along as the media content type is supported on cloud api side to get the returned media url.
